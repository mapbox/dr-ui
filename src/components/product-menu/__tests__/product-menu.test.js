import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './product-menu-test-cases.js';

describe('product-menu', () => {
  describe(testCases.mapboxMenu.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.mapboxMenu;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.locationMenu.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.locationMenu;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.arbitraryElement.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.arbitraryElement;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.customStyle.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.customStyle;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.beta.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.beta;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.betaLong.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.betaLong;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
