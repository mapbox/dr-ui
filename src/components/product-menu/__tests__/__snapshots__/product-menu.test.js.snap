// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-menu Basic renders as expected 1`] = `
<div
  className="dr-ui--product-menu"
>
  <a
    className="txt-fancy txt-l block color-blue-on-hover"
    href="/dr-ui/"
  >
    Mapbox product menu
  </a>
</div>
`;

exports[`product-menu Beta product renders as expected 1`] = `
<div
  className="dr-ui--product-menu"
>
  <div
    className="ml-neg3"
  >
    <div
      className="inline-block"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
    >
      <div
        aria-describedby="tooltip-1"
        className="txt-bold px6 round inline-block cursor-default txt-xs"
        style={
          Object {
            "background": "#f1f3fd",
            "borderColor": "#0428c8",
            "color": "#0c248d",
          }
        }
      >
        Beta
      </div>
      <div
        className="hide-visually"
        id="tooltip-1"
        role="tooltip"
      >
        <div
          className="txt-s wmax120"
        >
          This feature is in public beta and is subject to changes.
        </div>
      </div>
    </div>
  </div>
  <a
    className="txt-fancy txt-l block color-blue-on-hover"
    href="/vision/"
  >
    Vision SDK for Android
  </a>
</div>
`;

exports[`product-menu Beta product renders as expected 2`] = `
<div
  className="dr-ui--product-menu"
>
  <div
    className="ml-neg3"
  >
    <div
      className="inline-block"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
    >
      <div
        aria-describedby="tooltip-2"
        className="txt-bold px6 round inline-block cursor-default txt-xs"
        style={
          Object {
            "background": "#f1f3fd",
            "borderColor": "#0428c8",
            "color": "#0c248d",
          }
        }
      >
        Beta
      </div>
      <div
        className="hide-visually"
        id="tooltip-2"
        role="tooltip"
      >
        <div
          className="txt-s wmax120"
        >
          This feature is in public beta and is subject to changes.
        </div>
      </div>
    </div>
  </div>
  <a
    className="txt-fancy txt-l block color-blue-on-hover"
    href="/vision/"
  >
    Navigation SDK for Android
  </a>
</div>
`;

exports[`product-menu Custom trigger style - display only renders as expected 1`] = `
<div
  className="bg-gray-dark py6"
>
  <div
    className="dr-ui--product-menu"
  >
    <a
      className="txt-fancy txt-l block color-white color-gray-light-on-hover"
      href="/api-documentation/"
    >
      Light on dark
    </a>
  </div>
</div>
`;

exports[`product-menu Fake items, but one matches this test case's location renders as expected 1`] = `
<div
  className="dr-ui--product-menu"
>
  <a
    className="txt-fancy txt-l block color-blue-on-hover"
    href="/api-documentation/"
  >
    Items in here
  </a>
</div>
`;

exports[`product-menu Product menu with something arbitrary in it renders as expected 1`] = `
<div
  className="dr-ui--product-menu"
>
  <a
    className="txt-fancy txt-l block color-blue-on-hover"
    href="/api-documentation/"
  >
    Pizza pie
  </a>
</div>
`;
